flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -215,7 +218,12 @@ public void setNumberOfExecutionRetries(int numberOfExecutionRetries) {
 	 * @return The number of times the system will try to re-execute failed tasks.
 	 */
 	public int getNumberOfExecutionRetries() {
-		return numExecutionRetries;
+		int retries = executionConfig.getNumberOfExecutionRetries();
+		if (retries < -1) {
```

Comment:
this check shouldn't be necessary, since setNumberOfExecutionRetries already checks for it.